pkg/cloud/services/networking/securitygroups.go

```diff
@@ -412,11 +413,7 @@ func (s *Service) GetSecurityGroups(securityGroupParams []infrav1.SecurityGroupP
 		if err != nil {
 			return nil, err
 		}
-
-		if len(SGList) == 0 {
-			return nil, fmt.Errorf("security group %s not found", sg.Name)
-		}
```

Comment:
Note this change in behavior for GetSecurityGroups!
It used to return an error if none were found. Now it simply returns an empty slice.
If we want to keep the current behavior, we will need a way to tell the difference between "unable to get security groups" (e.g. network errors) and "these params doesn't match anything". Otherwise we have no way of knowing if we should create a new port or retry again later.